### PR TITLE
DBZ-5480 Augment a uniqueness key filed/value in regex topic naming strategy

### PR DESCRIPTION
**Pull request by harveyyue**
https://issues.redhat.com/browse/DBZ-5480

### CODE REVIEW

**Comment by harveyyue** [comment]
It's irrelevant test case failure.

2022-08-05T07:54:55.4342601Z [ERROR] Tests run: 28, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 384.391 s <<< FAILURE! - in io.debezium.connector.mysql.ReadOnlyIncrementalSnapshotIT
2022-08-05T07:54:55.4345133Z [ERROR] io.debezium.connector.mysql.ReadOnlyIncrementalSnapshotIT.testStopSnapshotKafkaSignal  Time elapsed: 51.025 s  <<< FAILURE!
2022-08-05T07:54:55.4346265Z org.junit.ComparisonFailure: expected:<[fals]e> but was:<[tru]e>
2022-08-05T07:54:55.4346975Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2022-08-05T07:54:55.4347859Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
2022-08-05T07:54:55.4348852Z 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2022-08-05T07:54:55.4349729Z 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
2022-08-05T07:54:55.4350418Z 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
2022-08-05T07:54:55.4351105Z 	at org.fest.assertions.ConstructorInvoker.newInstance(ConstructorInvoker.java:36)
2022-08-05T07:54:55.4352036Z 	at org.fest.assertions.ComparisonFailureFactory.newComparisonFailure(ComparisonFailureFactory.java:60)
2022-08-05T07:54:55.4352941Z 	at org.fest.assertions.ComparisonFailureFactory.comparisonFailure(ComparisonFailureFactory.java:46)
2022-08-05T07:54:55.4353692Z 	at org.fest.assertions.Fail.comparisonFailed(Fail.java:83)
2022-08-05T07:54:55.4354272Z 	at org.fest.assertions.Fail.failIfNotEqual(Fail.java:71)
2022-08-05T07:54:55.4354732Z 	at org.fest.assertions.GenericAssert.isEqualTo(GenericAssert.java:217)
2022-08-05T07:54:55.4355278Z 	at org.fest.assertions.BooleanAssert.isEqualTo(BooleanAssert.java:73)
2022-08-05T07:54:55.4355798Z 	at org.fest.assertions.BooleanAssert.isFalse(BooleanAssert.java:63)
2022-08-05T07:54:55.4356530Z 	at io.debezium.connector.mysql.ReadOnlyIncrementalSnapshotIT.lambda$testStopSnapshotKafkaSignal$10(ReadOnlyIncrementalSnapshotIT.java:292)
2022-08-05T07:54:55.4357312Z 	at org.awaitility.core.CallableCondition$ConditionEvaluationWrapper.eval(CallableCondition.java:99)
2022-08-05T07:54:55.4357907Z 	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:203)
2022-08-05T07:54:55.4358435Z 	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:190)
2022-08-05T07:54:55.4358905Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-08-05T07:54:55.4359446Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-08-05T07:54:55.4360027Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-08-05T07:54:55.4360477Z 	at java.base/java.lang.Thread.run(Thread.java:833)

**Comment by Naros** [comment]
Logged [DBZ-5483](https://issues.redhat.com/browse/DBZ-5483) to look into the test failure.

File: debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java [hunk 1]
```diff
@@ -35,6 +36,30 @@
 
     String transactionTopic();
 
+    /**
```
Comment (jpechane) [inline debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java:39]: @harveyyue I don't think these settings belong here

1) The comments clearly state that it is for regex only so it shold be pulled-down there
2) The class is named Topic*Naming*Strategy and these settings are completely unrelated to topic naming

Comment (harveyyue) [reply debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java:39]: Move these settings to AbstractRegexTopicNamingStrategy.

File: debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java [hunk 2]
```diff
@@ -134,6 +135,13 @@ public TableSchema create(TopicNamingStrategy topicNamingStrategy, Table table,
             addField(keySchemaBuilder, table, column, null);
             hasPrimaryKey.set(true);
         });
+        if (AbstractRegexTopicNamingStrategy.class.isAssignableFrom(topicNamingStrategy.getClass())) {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java:138]: @harveyyue I am not a fan of having the code placed here. Can't we provide a method on the naming strategy that will be called so it can do necessary changes?

Comment (harveyyue) [reply debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java:138]: @jpechane  These code changes will depend on connect-api, so I didn't provide a mehod in TopicNamingStrategy interface.

Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java:138]: @harveyyue I see, still there can be default no-op method on the interface itself. Maybe we could pass a callback lambda/method handle that would be called.

Comment (harveyyue) [reply debezium-core/src/main/java/io/debezium/relational/TableSchemaBuilder.java]: @jpechane I have implemented it, please help to review again.

File: debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java [hunk 3]
```diff
@@ -66,4 +66,33 @@ default String sanitizedTopicName(String topicName) {
...
+    default TopicValueAugment keyValueAugment() {
+        return NO_VALUE_OP;
+    }
+
+    /**
+     * An interface that augments the field to topic key/value schema.
+     * @param <S> Connect schema builder
+     */
+    interface TopicSchemaAugment<S> {
+        boolean augment(S schemaBuilder);
```
[the remark this answers is not in the document]
Comment (harveyyue) [reply debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java:83]: @jpechane ok, will implement it.

[the remark this answers is not in the document]
Comment (harveyyue) [reply debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java:88]: @jpechane done.

**Comment by jpechane** [comment]
@harveyyue Applied, thanks!
